src/components/common/CSV.js

```diff
@@ -3,19 +3,58 @@ import { CSVLink } from 'react-csv';
 import PropTypes from 'prop-types';
 import styles from '../chart-scroll/ChartScroll.css';
 import downloadImage from '../../assets/download6.png';
+import Modal from 'react-modal';
 
 const CSV = ({ data, filename }) => {
+  const [modalIsOpen, setIsOpen] = React.useState(false);
+  function openModal() {
+    setIsOpen(true);
+  }
+  function closeModal(){
+    setIsOpen(false);
+  }
   return (
-    <div className={styles.csv}>
-      <CSVLink 
-        data={[...data]}
-        filename={filename}
-        target='_blank'>
-        <p className={styles.downloadContainer}>
-          <img className={styles.downloadImage} src={downloadImage} alt='download' />
-          <span className={styles.toolTip}>Download a CSV of this data</span>
-        </p>
-      </CSVLink>
+    <div>
+      {/* <img className={styles.downloadImage} src={downloadImage} alt='download' /> */}
+      <div >
+        <button 
+          onClick={openModal}
+          className={styles.downloadHider}>
+          <img 
+            className={styles.downloadImage}
+            src={downloadImage}
+          />
+          <p className={styles.downloadContainer}>
+            <span className={styles.toolTip}>Download a CSV of this data</span>
+          </p>
+        </button>
+      </div>
+      <Modal
+        isOpen={modalIsOpen}
+        className={styles.modal}
+      >
+        <div>
+          <p className={styles.csv}>
+            <CSVLink 
+              data={[...data]}
+              filename={filename}
+              target='_blank'>
+            download complete data
+              <br/>
+            </CSVLink>
+          </p>
+         
+          <p className={styles.csv}>
+            <CSVLink 
+              data={[...data].slice(-1)}
```

Comment:
I think this is the problem with the age range data. It will probably be the same with agency data too....
I'm not sure if there's a super quick fix. May have to do something with the route of the aggregation. I'll think on it.